### PR TITLE
Add latest version to the home page cards for each project

### PR DESCRIPTION
**Pull request by rambo0247**
Solves https://github.com/neoforged/project-listing/issues/7

### CODE REVIEW

File: src/components/ProjectIndex.vue [hunk 1]
```diff
@@ -33,13 +37,41 @@
...
-  data() {
+  async setup() {
+    const repos = ref([])
+
+    const fetchRepoData = async () => {
+      const repoDataPromises = Object.keys(json).map(async (project) => {
+        const projectInfo = json[project];
+        const mavenPath = projectInfo.artifact.replace(/\./g, '/').replace(':', '/');
+
+        const versions = await fetch(`https://maven.neoforged.net/api/maven/versions/releases/${mavenPath}`)
```
Comment (Matyrobbrt) [inline src/components/ProjectIndex.vue:52]: For the latest version the path `/api/maven/latest/version/{repository}/{gav}` can be used to save bandwidth.

File: src/components/ProjectIndex.vue [hunk 2]
```diff
@@ -1,26 +1,30 @@
...
-                <RepoDescription :repo="repo" />
+                <RepoDescription :repo="repo.projectRepo" />
               </Suspense>
             </v-card-item>
+
             <v-card-actions>
-              <v-btn :href="`/${repo}`">Details</v-btn>
+              <v-btn :href="`/${repo.projectRepo}`">Details</v-btn>
+              <v-spacer></v-spacer>
+              <v-btn :href="`/${repo.projectRepo}`">{{ repo.latestVersion || 'N/A' }}</v-btn>
```
Comment (Matyrobbrt) [inline src/components/ProjectIndex.vue:27]: I don't like the N/A, just make it show nothing. Additionally, the use of a button is quite weird as it changes the font, it should be simple text.

**Review by Matyrobbrt — CHANGES_REQUESTED** [review]

File: src/components/ProjectIndex.vue (hunk 1 above)
Comment (rambo0247) [reply src/components/ProjectIndex.vue:52]: > For the latest version the path `/api/maven/latest/version/{repository}/{gav}` can be used to save bandwidth.

do you mean `/api/maven/latest/version/releases/{artifact}` instead?

Comment (rambo0247) [reply src/components/ProjectIndex.vue]: never mind, it's done

**Review by Matyrobbrt — APPROVED** [review]

**Review by Matyrobbrt — APPROVED** [review]

**Pull request merged**